src/app/actions/SearchActions.js

```diff
@@ -12,26 +13,70 @@ export const searchResults = (results) => {
   };
 };
 
-export const search = (query, field = 'q') => {
-  const appEnv = process.env.APP_ENV || 'production';
+export const workDetail = (item) => {
+  return {
+    type: Actions.FETCH_WORK,
+    item,
+  };
+};
+
+const appEnv = process.env.APP_ENV || 'production';
+const apiUrl = appConfig.api[appEnv];
+const { searchPath, recordPath } = appConfig.api;
+const searchUrl = apiUrl + searchPath;
+
+export const searchGet = (query) => {
   // Need a parsed query input to use for each filter
   const userQuery = (query) ? encodeURIComponent(query) : '*';
-  // Need a client to send the search and receive results
-  // Need to pass the results to a renderer
-  const apiUrl = appConfig.api[appEnv];
 
   return (dispatch) => {
-    return axios.get(apiUrl, { params: { q: userQuery } })
+    return axios.get(searchUrl, { params: { q: userQuery } })
       .then((resp) => {
         if (resp.data) {
           dispatch(searchResults(resp.data));
         }
       })
       .catch((error) => {
-        console.log('Error communicating with Elasticsearch', apiUrl, userQuery, error);
-        throw new Error('Error communicating with Elasticsearch', apiUrl, userQuery, error);
+        console.log('An error occurred during searchGet', error);
+        throw new Error('An error occurred during searchGet', error);
+      });
+  };
+};
+
+export const searchPost = (query, field) => {
+  // Need a parsed query input to use for each filter
+  const userQuery = (query) ? encodeURIComponent(query) : '*';
+
+  return (dispatch) => {
+    return axios.post(searchUrl, { queries: [{ field, value: userQuery }] })
+      .then((resp) => {
+        if (resp.data) {
+          dispatch(searchResults(resp.data));
+        }
+      })
+      .catch((error) => {
+        console.log('An error occurred during searchPost', error);
+        throw new Error('An error occurred during searchPost', error);
+      });
+  };
+};
+
+export const fetchWork = (workId) => {
+  const recordUrl = apiUrl + recordPath;
+  return (dispatch) => {
+    return axios.get(recordUrl, { params: { recordID: workId } })
+      .then((resp) => {
+        if (resp.data) {
+          dispatch(workDetail(resp.data));
+          // Do I need to change state of the searchResults here?
```

Comment:
Hmm why do you need to change the state for `searchResults` if you are fetching on specific work? To answer the question, no, you would update the state in the reducer and not in the action.

Reply:
This shows my lack of React knowledge and understanding since I wasn't able to just show the details without also showing the results. It's unclear to me how "pages" are handled in React where one display only shows one set of props from state while hiding other props from state. Perhaps I need to switch containers?